src/BugsnagUnity/Delivery.cs

```diff
@@ -524,21 +524,53 @@ private IEnumerator ProcessCachedItems(Type t)
 
                     try
                     {
+                        // if something goes wrong at this stage then we silently discard the file as it's most likely that the file wasn't fully serialised to disk
                         payloadDictionary = ((JsonObject)SimpleJson.DeserializeObject(payloadJson)).GetDictionary();
                     }
-                    catch (Exception e)
+                    catch
                     {
-                        Debug.LogException(new Exception("Bugsnag Error. Deserialising a cached payload for delivery failed: " + e.Message + " Cached json: " + payloadJson));
+                        if (isSession)
+                        {
+                            _cacheManager.RemoveCachedSession(id);
+                        }
+                        else
+                        {
+                            _cacheManager.RemoveCachedEvent(id);
+                        }
                         continue;
                     }
 
+
+                    // if something goes wrong at this stage then we discard the file and report the error as it might be a bug in the sdk
```

Comment:
Probably just drop this comment now? The comment below suggests the future improvement.